File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int n : nums) {
            freq.put(n, freq.getOrDefault(n, 0) + 1);
        }

        List<Integer> keys = new ArrayList<>(freq.keySet());
        // System.out.println(keys);
        keys.sort((a, b) -> freq.get(b) - freq.get(a));

        int[] res = new int[k];

        for (int i = 0; i < k; i++) {
            res[i] = keys.get(i);
        }

        return res;
    }
}

File: Data Structures & Algorithms/special-array-i/submission-1.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        
        if len(nums) <= 1:
            return True
        
        l ,r = 0, 1

        while r < len(nums):
            if nums[l]%2 == 0 and nums[r] % 2 == 0:
                return False
            if nums[l]%2 == 1 and nums[r]% 2 ==1:
                return False

            l += 1
            r += 1
        
        return True